File: android/VIRGINSApp/app/src/main/java/app/virgins/data/model/Message.kt
package app.virgins.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Conversation(
    @Json(name = "_id") val id: String,
    val participants: List<String>,
    val partner: MatchedUser? = null,
    val lastMessage: String? = null,
    val lastMessageAt: String? = null,
    val unreadCount: Int = 0,
    val isActive: Boolean = true
)

@JsonClass(generateAdapter = true)
data class ChatMessage(
    @Json(name = "_id") val id: String,
    val conversationId: String,
    val senderId: String,
    val content: String,
    val type: String = "text",
    val readAt: String? = null,
    val createdAt: String,
    val isDeleted: Boolean = false
)

@JsonClass(generateAdapter = true)
data class SendMessageRequest(
    val content: String,
    val type: String = "text"
)

File: ios/VIRGINSApp/Sources/Models/Message.swift
import Foundation

struct Conversation: Codable, Identifiable {
    let id: String
    let matchId: String
    let partner: MatchedUser
    var lastMessage: String?
    var lastMessageAt: Date?
    var unreadCount: Int

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case matchId, partner, lastMessage, lastMessageAt, unreadCount
    }
}

struct ChatMessage: Codable, Identifiable {
    let id: String
    let conversationId: String
    let senderId: String
    var content: String
    let type: MessageType
    var readAt: Date?
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case conversationId, senderId, content, type, readAt, createdAt
    }

    enum MessageType: String, Codable {
        case text, image, voice, gif
    }

    func isSentBy(_ userId: String) -> Bool { senderId == userId }
}
